src/clj/game/core/runs.clj

```diff
@@ -195,47 +211,84 @@
+        ]
+    ;; Steal costs are additional costs and can be denied by the runner.
+    (continue-ability state :runner
+                      (if (> (count choices) 1)
+                        ;; Use the better function for multiple costs
+                        (steal-pay-choice state :runner choices '() n c)
+                        ;; Otherwise, just handle everything right friggin here
+                        {:delayed-completion true
+                         :prompt prompt-str
+                         :choices (into [] (concat ability-strs
+                                                   steal-str
+                                                   dont-pay-str))
+                         :effect (req (cond
+                                        ;; Can't steal
+                                        (= target "OK")
+                                        (do (trigger-event state side :no-steal c)
+                                            (resolve-steal-events state side eid c))
+
+                                        ;; Steal normally
+                                        (= target "Steal")
+                                        (resolve-steal state :runner eid c)
+
+                                        ;; Pay single additiional cost to steal
+                                        (= target "Pay")
+                                        (when-completed (pay-sync state side nil cost {:action :steal-cost})
+                                                        (do (system-msg
+                                                              state side
+                                                              (str "pays " cost-as-symbol " to steal " card-name))
+                                                            (resolve-steal state side eid c)))
+
+                                        ;; Don't pay single additional cost to steal
+                                        (.contains target "Don't")
+                                        (do (trigger-event state side :no-steal c)
+                                            (effect-completed state side eid))
+
+                                        ;; Use trash ability
+                                        (.contains target "ability")
+                                        (let [idx (.indexOf ability-strs target)
+                                              trash-ab-card (nth trash-ab-cards idx)
+                                              cdef (-> (card-def trash-ab-card)
+                                                       :interactions
+                                                       :trash-ability
+                                                       (assoc :eid eid))]
+                                          (resolve-ability state side cdef trash-ab-card [c])
+                                          (trigger-event state side :no-steal c)
+                                          (effect-completed state side eid))))})
```

Comment:
Don't need this, as the `eid` has been passed to the cdef you are resolving above.

Reply:
Ah, perfect. Thanks!